### ISSUE TITLE
descriptor loss

### ISSUE THREAD

**Issue by Hjy0801** [cvg/SOLD2#78]
Hello, when i trained the full pipeline, the descriptor loss can not decreased? it was about 1 all time. why is this? Can you help me? Thank you.

**Comment by rpautrat** [comment]
Hi, can you share the tensorboard curves of the different losses? Are the other losses decreasing or not?
On which dataset are you training also?

**Comment by rpautrat** [comment]
Did you change anything in the code of the main branch?
Did you try visualizing the exported dataset with [this notebook](https://github.com/cvg/SOLD2/blob/main/notebooks/visualize_exported_dataset.ipynb) to check that everything was fine?

**Comment by Hjy0801** [comment]
I did't change any code and check the exported dataset. I operated the command of step 5 and used the sold2_synthetic.tar as the pre-train model.
![image](https://user-images.githubusercontent.com/127933270/225279488-e731dd30-c2f6-468a-a242-1ae892bb2c09.png)
This is the train process. descriptor_loss always is 1.
![image](https://user-images.githubusercontent.com/127933270/225279880-2a7cab21-7057-4b57-bd2e-357c6cd58eba.png)
This is the train datasets.

**Comment by rpautrat** [comment]
The exported dataset looks good.

Did you modify the field 'return_type' in config/wireframe_dataset.yaml to 'paired_desc', as requested in the ReadMe?

**Comment by Hjy0801** [comment]
yes,I did

**Comment by rpautrat** [comment]
In our original work, we first did step 4 of the Readme, before proceeding to step 5. But as I understand it, you skipped step 4 (i.e. training the detector only on the Wireframe dataset), right? Maybe that is why the descriptor loss is stuck, you need to pre-train the network with step 4, then fine-tune it with the descriptor in step 5.

The value of 1 displayed in your descriptor loss is basically the margin used in the triplet loss. In my experience, it is common that when training a network with triplet loss, the descriptor loss can stay stuck at this margin value (1 in your case) for a while. But if you keep training, the network eventually manage to improve and to go below this margin value.

So as a conclusion, I would first pre-train the network with step 4, then try step 5 and train long enough to let time to the network to go below the margin value.

**Comment by Hjy0801** [comment]
Thanks for your explanation,. I will do the step 4 and then do the step 5. Have a good week!
